GreeterTest.java

```diff
@@ -0,0 +1,29 @@
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+/*
+ * thanks to dfa at  http://stackoverflow.com/a/1119559
+ */
+public class GreeterTest{
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @Before
+    public void setUpStream(){
+        System.setOut(new PrintStream(outContent));
+    }
+    @After
+    public void cleanUpStream(){
+        System.setOut(null);
```

Comment:
This "breaks" System.out for any subsequent use.  Each test should set up data it needs, run the test, then restore everything to the way it was before.